North-Seattle-College/ad440-winter2021-thursday-repo
# Implement Delete User From All Users Page via Delete Option

## This PR is for TASK [238](https://github.com/North-Seattle-College/ad440-winter2021-thursday-repo/issues/238)
- The UI should be updated so we can delete a specific user on the user's webpage.

# How to test it: 

1. Navigate to the correct directory:
`- cd ui/react-ui`
2. Install all packages, modules, and dependencies:
`- npm install `
3. Start the app in the development model:
`- npm start `

- After you start npm start you should see the following: 
![image](https://user-images.githubusercontent.com/32443542/111101320-e1d40a80-8506-11eb-9fc9-0fe959854a74.png)


-  Next Step:
- [ ] Navigate to `/users`

- Next Step: 
- [ ] After all users have been loaded you should see the new **Remove User** with **Delete Option**
![image](https://user-images.githubusercontent.com/32443542/111101565-6b83d800-8507-11eb-8092-3d84951a1b3e.png)

- Next Step:
- [ ] Select the user you want to delete [ look at the row and click on the delete field]  Verify that you can see a pop-up that includes the first and lastname of the user you want to delete. The pop-up includes a Cancel and Confirm option. 
- 
![image](https://user-images.githubusercontent.com/32443542/111101978-404db880-8508-11eb-9517-6f4819d40661.png)
   


- When you select **Cancel** the pop-up closes and nothing happens. When you select **Confirm** you will see a pop-up on the right top that shows a confirmation if the user got deleted successfully, if not it shows an error message

![image](https://user-images.githubusercontent.com/32443542/111100704-8ce3c480-8505-11eb-9d5c-c6e9af63079f.png)

- You can verify that the user no longer exists in the DB.

# TIME SPEND: 

DATE | ACTIVITY | TIME
-- | -- | --
03/11/21 | Reviewed UI Features and started implementing Delete Options | 2hr
03/13/21 | Fixed modal pop up| 2 Hr
03/13/21 | Encountered an issue with delete method | 1hr
03/13/21 | Worked on fixing delete request for HTTPTriggerUsersUserId | ~1hr
03/14/21 | Worked on Delete issue with HTTP Trigger and fixed it  | 2hr
03/14/21 | Created this PR | ~1hr
total| |9 Hr
